File: src/main/java/github/macro/pieces/Emperor.java
package github.macro.pieces;

import github.macro.Direction;
import github.macro.players.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Macro303 on 2021-Jun-02.
 */
public class Emperor extends Piece {
	public Emperor(@NotNull Player player) {
		super(player, 4, "E", List.of(
				Direction.NORTH,
				Direction.NORTH_EAST,
				Direction.EAST,
				Direction.SOUTH_EAST,
				Direction.SOUTH,
				Direction.SOUTH_WEST,
				Direction.WEST,
				Direction.NORTH_WEST
		));
	}

	@Override
	public String toString() {
		return "Emperor{} " + super.toString();
	}
}

File: src/main/kotlin/github/macro/pieces/Emperor.kt
package github.macro.pieces

import github.macro.Direction
import github.macro.players.Player

/**
 * Created by Macro303 on 2018-02-08.
 */
class Emperor(player: Player) : Piece(
	player = player,
	distance = 4,
	symbol = "E",
	directions = arrayOf(
		Direction.NORTH,
		Direction.NORTH_EAST,
		Direction.EAST,
		Direction.SOUTH_EAST,
		Direction.SOUTH,
		Direction.SOUTH_WEST,
		Direction.WEST,
		Direction.NORTH_WEST
	)
) {
//	private static final String symbol = "\u265A";
//	private static final String symbol = "♚";
//	private static final String symbol = "E";

	override fun toString(): String = "Emperor() ${super.toString()}"
}